nucypher/config/base.py

```
@@ -0,0 +1,216 @@
+        :param modifier: A unique string to modify the filename if the file already exists.
+        :param override: Allow generation of an existing filepath.
+        :return: The generated filepath.
+
+        """
+        if not filepath:
+            filename = self.generate_filename()
+            filepath = os.path.join(self.config_root, filename)
+        if os.path.exists(filepath) and not override:
+            if not modifier:
+                raise FileExistsError(f"{filepath} exists and no filename modifier supplied.")
+            filename = self.generate_filename(modifier=modifier)
+            filepath = os.path.join(self.config_root, filename)
+        self.filepath = filepath
+        return filepath
+
+    def to_configuration_file(self, filepath: str = None, modifier: str = None, override: bool = False) -> str:
+        filepath = self.generate_filepath(filepath=filepath, modifier=modifier, override=override)
+        filepath = self._write_configuration_file(filepath=filepath, override=override)
+        return filepath
+
+    @classmethod
+    def from_configuration_file(cls, filepath: str = None, **overrides) -> 'BaseConfiguration':
+        filepath = filepath or cls.default_filepath()
+        payload = cls._read_configuration_file(filepath=filepath)
+        instance = cls(filepath=filepath, **payload, **overrides)
+        return instance
+
+    @classmethod
+    def _read_configuration_file(cls, filepath: str) -> dict:
+        """Reads `filepath` and returns the deserialized JSON payload dict."""
+        with open(filepath, 'r') as file:
+            raw_contents = file.read()
+            payload = cls.deserialize(raw_contents)
+        return payload
+
+    def _write_configuration_file(self, filepath: str, override: bool = False) -> str:
+        """Writes to `filepath` and returns the written filepath.  Raises `FileExistsError` if the file exists."""
+        if os.path.exists(filepath) and not override:
```

What is "override" here?
Equivalent to "fail silently"?

---

`override` allows the file to be written if it exists.